src/common/format.ts

```diff
@@ -10,7 +10,8 @@ export const displayNameForDeletedClient = "Deleted Client";
 
 export const displayNameForNullDriverName = "Unknown Driver";
 
-export const phoneNumberRegex = /^([+]?[(]?[0-9]{3}[)]?[-\s.]?[0-9]{3}[-\s.]?[0-9]{4,6})?$/;
+// Following characters excluded from regex as are removed before checking format matches: ( ) - \s
+export const phoneNumberRegex = /^([0][0-9]{9,12}|[+][0-9]{7,15})?$/;
```

Comment:
This is already a much simpler regex than before thanks to pre-processing, but I think there are some tweaks we can make for it to be even better.
1. Handle the +44 code explicitly. It's much more likely that a user types a +44 number than any other country code, so I think we want to handle it as precisely as possible.
2. For regex-cleanliness, we can avoid using square brackets for single characters, which can just be by themselves, and `[0-9]` can be shorthanded as `\d`.
3. As far as I can tell, UK phone numbers are always either nine or ten digits after the 0 or +44 ([source](https://en.wikipedia.org/wiki/Telephone_numbers_in_the_United_Kingdom#Format)), so I think we can tighten that range a bit.

Putting this all together, I've come up with the regex:
`/^((0|\+44)\d{9,10}|\+(?!44)\d{7,15})?$/`
Have a play around with it on [regex101.com](regex101.com) to check you understand what it's doing, and let me know if I've misunderstood anything or if you can think of other improvements.